server_env_magentoerpconnect/magento_model.py

```diff
@@ -31,41 +30,52 @@
 _logger = logging.getLogger(__name__)
 
 
-class magento_backend(orm.Model):
+class magento_backend(models.Model):
     _inherit = 'magento.backend'
 
-    def _get_environment_config_by_name(self, cr, uid, ids, field_names,
-                                        arg, context=None):
-        values = {}
-        for backend in self.browse(cr, uid, ids, context=context):
-            values[backend.id] = {}
-            for field_name in field_names:
-                section_name = '.'.join((self._name.replace('.', '_'),
-                                         backend.name))
+    location = fields.Char(
+        compute='_get_environment_config_by_name',
+        string='Location',
+        readonly=True,
+        required=False,
+    )
+    username = fields.Char(
+        compute='_get_environment_config_by_name',
+        string='Username',
+        readonly=True,
+        required=False,
+        )
+    password = fields.Char(
+        compute='_get_environment_config_by_name',
+        string='Password',
+        readonly=True,
+        required=False,
+    )
+
+    @api.model
+    def _get_env_fields(self):
+        """Return the list of fields that are concerned
+        by the environment setup
+
+        :return: a list of field name
+        :rtype: list
+        """
+
+        return ['password', 'username', 'location']
+
+    @api.one
+    def _get_environment_config_by_name(self):
+        """Compute the fields values for current environment"""
+        for backend in self:
+            for field_name in self._get_env_fields():
+                section_name = '.'.join(
+                    (self._name.replace('.', '_'), backend.name)
+                )
                 try:
                     value = serv_config.get(section_name, field_name)
-                    values[backend.id][field_name] = value
+                    self[field_name] = value
```

Comment:
maybe I am wrong but it should be "backend[field_name]" instead of self[field_name]